### PR TITLE
Add define rec fun to cvc parser

### PR DESCRIPTION
**Pull request by arjunvish**
CVC Front end for recursive functions (Issue #1316)

### CODE REVIEW

File: src/parser/cvc/Cvc.g [hunk 1]
```diff
@@ -703,10 +703,19 @@ mainCommand[std::unique_ptr<CVC4::Command>* cmd]
   Expr f;
   SExpr sexpr;
   std::string id;
-  Type t;
+  Type t,t2,t3;
```
Comment (mpreiner) [inline src/parser/cvc/Cvc.g]: `t2` and `t3` seem to be unused.

File: src/parser/cvc/Cvc.g [hunk 2]
```diff
@@ -880,6 +889,69 @@ mainCommand[std::unique_ptr<CVC4::Command>* cmd]
...
+    }
+    (COMMA{
+      formCommaFlag=true;
+    })?
+    {
+      if( f.getKind()==kind::LAMBDA ){
+        for( unsigned i=0,size=f[0].getNumChildren(); i<size; i++)
+        {
+          bvs.push_back( f[0][i] );
+        }
```
Comment (mpreiner) [inline src/parser/cvc/Cvc.g]: You can use `bvs.insert{bvs.end(), f[0].begin(), f[0].end())` instead of the for loop.

File: src/parser/cvc/Cvc.g [hunk 3]
```diff
@@ -880,6 +889,69 @@ mainCommand[std::unique_ptr<CVC4::Command>* cmd]
...
+      if(idCommaFlag){
+        PARSER_STATE->parseError("Cannot end function list with comma");
+      }
+      if(formCommaFlag){
+        PARSER_STATE->parseError("Cannot end function definition list with comma");
+      }
+      if(funcs.size()!=formulas.size()){
+        PARSER_STATE->parseError("Number of functions doesn't match number of function definitions");
+      }
+      for(unsigned int i = 0; i < funcs.size(); i++){
```
Comment (mpreiner) [inline src/parser/cvc/Cvc.g]: Cache `funcs.size()`, e.g. `for(unsigned int i = 0, size = funcs.size(); i < size; i++)`

File: src/parser/cvc/Cvc.g [hunk 4]
```diff
@@ -880,6 +889,70 @@ mainCommand[std::unique_ptr<CVC4::Command>* cmd]
...
+    }
+    (COMMA{
+      formCommaFlag=true;
+    })?
+    {
+      if( f.getKind()==kind::LAMBDA ){
+        /*for( unsigned i=0,size=f[0].getNumChildren(); i<size; i++)
+        {
+          bvs.push_back( f[0][i] );
+        }*/
```
Comment (mpreiner) [inline src/parser/cvc/Cvc.g]: Remove the unused code.

Comment (arjunvish) [reply src/parser/cvc/Cvc.g]: Silly overlooking, sorry.

**Pull request merged**